lib/rubocop/cop/style/fetch_env_var.rb

```diff
@@ -112,11 +112,22 @@ def used_if_condition_in_body(node)
         end
 
         def used_in_condition?(node, condition)
-          if condition.send_type? && (!condition.comparison_method? && !condition.predicate_method?)
+          if condition.send_type? &&
+             (!condition.comparison_method? &&
+              !condition.predicate_method? &&
+              !condition.assignment_method?)
             return false
           end
 
-          condition.child_nodes.any?(node)
+          used_in_condition_with_assignment?(node, condition) || condition.child_nodes.any?(node)
+        end
+
+        def used_in_condition_with_assignment?(node, condition)
+          if condition.send_type? && condition.assignment_method?
+            node.child_nodes == node.child_nodes & condition.child_nodes
```

Comment:
Is this condition necessary? If the condition can be eliminated, the following looks simple.

```diff
diff --git a/lib/rubocop/cop/style/fetch_env_var.rb b/lib/rubocop/cop/style/fetch_env_var.rb
index 06b3fdfc4..5e8bb5247 100644
--- a/lib/rubocop/cop/style/fetch_env_var.rb
+++ b/lib/rubocop/cop/style/fetch_env_var.rb
@@ -112,22 +112,12 @@ module RuboCop
         end

         def used_in_condition?(node, condition)
-          if condition.send_type? &&
-             (!condition.comparison_method? &&
-              !condition.predicate_method? &&
-              !condition.assignment_method?)
-            return false
+          if condition.send_type?
+            return true if condition.assignment_method?
+            return false if !condition.comparison_method? && !condition.predicate_method?
           end

-          used_in_condition_with_assignment?(node, condition) || condition.child_nodes.any?(node)
-        end
-
-        def used_in_condition_with_assignment?(node, condition)
-          if condition.send_type? && condition.assignment_method?
-            node.child_nodes == node.child_nodes & condition.child_nodes
-          else
-            false
-          end
+          condition.child_nodes.any?(node)
         end

         def offensive?(node)
```

So just update `used_in_condition?` method.

```ruby
         def used_in_condition?(node, condition)
           if condition.send_type?
             return true if condition.assignment_method?
             return false if !condition.comparison_method? && !condition.predicate_method?
           end

           condition.child_nodes.any?(node)
         end
```

Reply:
Thank you very much.
I've modified the conditional statement. However, the following conditional statement is still needed:

```ruby
node.child_nodes == node.child_nodes & condition.child_nodes
```

The following code cannot be considered offensive:

```ruby
if ENV['X'] = x
  puts ENV['Y']
end
```